Improve the Keda module testing strategy doc (#374)
kyma-project/keda-manager: docs/contributor/04-10-ci-cd.md

| Name | Required | Description |
|------|----------|-------------|
|[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are no broken links in the pull request `.md` files.|
|[`pre-keda-manager-operator-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/keda-manager.yaml#L43)|true|Builds Keda operator's image and pushes it to the `dev` registry.|
|[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository doesn't contain any vulnerabilities.|

> 

Reviewer: I would suggest to adding missing tests described [here](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L18-L56). I think three of them are missing:

```
Keda verify / lint
Keda verify / unit-test
Keda verify / integration-test
```

| Name | Required | Description |
|------|----------|-------------|

> |[`Keda verify / lint`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L19)|true|Is responsible for the Operator linting and static code analysis.|

Reviewer 2: ```suggestion
|[`Keda verify / lint`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L19)|true|Responsible for the Operator linting and static code analysis.|
```

> |[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are no broken links in the pull request `.md` files.|

Reviewer 2: ```suggestion
|[`PR Markdown Link Check / markdown-link-check (pull_request)`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/pr-markdown-link-check.yaml)|true|Checks if there are any broken links in `.md` files included in the pull request .|
```

|[`Keda verify / unit-test`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L29)|true|Runs basic unit tests of Operator's logic.|

> |[`Keda verify / integration-test`](https://github.com/kyma-project/keda-manager/blob/main/.github/workflows/keda-verify.yml#L38)|true|Runs the basic functionality integration test suite for the Operator in a k3d cluster.|

Reviewer 2: What `Operator` do you mean? Is it `Operator` or `the Operator`?

Author: It's Keda Operator, so I think it's `the`

Author: Actually, should be without `the` :D I've fixed that :)

|[`pre-keda-manager-operator-build`](https://github.com/kyma-project/test-infra/blob/main/templates/data/keda-manager.yaml#L43)|true|Builds Keda operator's image and pushes it to the `dev` registry.|

> |[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository doesn't contain any vulnerabilities.|

Reviewer 2: ```suggestion
|[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository contains any vulnerabilities.|
```

> |[`pre-keda-manager-presubmit-scanner`](https://github.tools.sap/kyma/security-scanners#readme)|true|Checks if the repository doesn't contain any vulnerabilities.|

Author: Should not, I'll fix it :)

> ## Pipelines Running on Main Branch 

Reviewer 2: ```suggestion
## Pipelines Running on `main` Branch 
```
